### PR TITLE
RST-10524 Use lz4 compression

### PR DESCRIPTION
**Pull request by nleblanc-lr**

### CODE REVIEW

File: rosbag_snapshot/src/snapshotter.cpp [hunk 1]
```diff
@@ -570,6 +573,7 @@ bool Snapshotter::triggerSnapshotCb(rosbag_snapshot_msgs::TriggerSnapshot::Reque
 
   // Create bag
   rosbag::Bag bag;
+  bag.setCompression(CompressionType::LZ4);
```
Comment (nleblanc-lr) [inline rosbag_snapshot/src/snapshotter.cpp:576]: This is the only real change. The rest were for roslint errors

**Review by AadityaRavindran — APPROVED** [review]

**Comment by MJohnson459** [comment]
What are the performance implications of this? The bump bags can be fairly large and I assume the CPU usage will go up with this? Although I suppose they may only seem large because they aren't compressed :D. It would be good to have a quick comparison at least.

**Comment by nleblanc-lr** [comment]
> What are the performance implications of this? The bump bags can be fairly large and I assume the CPU usage will go up with this? Although I suppose they may only seem large because they aren't compressed :D. It would be good to have a quick comparison at least.

I intend to do some testing before merging this. The compression is only performed when the message buffers are written to a bag file when a snapshot is triggered, so it will potentially be a spike in CPU usage and not a sustained increase. Just in local-sim, I didn't see any noticeable difference, but I'll test it on a bot.

**Review by MJohnson459 — APPROVED** [review]

**Comment by nleblanc-lr** [comment]
I did my own little performance test (capturing the output of `top`) and there doesn't seem to be any difference. I tested with `v2-19830` at Sunny Slope.

Created a bag without compression with 10273 messages. It took about 2 seconds to create and basically uses a core during that time:
v2-19830-no-compression

Created a bag with lz4 compression with 10315 messages. It also took about 2 seconds to create and the CPU usage looks essentially the same:
v2-19830-with-lz4

**Pull request merged**